File: scratch/check_syntax.py
import sys

with open(r'c:\Users\Lenovo Flex i7\Documents\Apps\KairosVisuals\src\app\quotations\create\page.tsx', 'r', encoding='utf-8') as f:
    content = f.read()

# Check braces
open_braces = content.count('{')
close_braces = content.count('}')
print(f"Braces: {open_braces} open, {close_braces} close")

# Check brackets
open_sq = content.count('[')
close_sq = content.count(']')
print(f"Square brackets: {open_sq} open, {close_sq} close")

# Check parens
open_par = content.count('(')
close_par = content.count(')')
print(f"Parens: {open_par} open, {close_par} close")

# List lines with possible issues (missing semicolons at end of blocks are usually fine in JS, but let's see)
lines = content.split('\n')
for i, line in enumerate(lines):
    if '<<' in line or '>>' in line:
        print(f"Line {i+1} might have conflict markers: {line}")

File: scratch/check_syntax.js
const fs = require('fs');
const content = fs.readFileSync('src/app/solicitud/page.tsx', 'utf8');

let braces = 0;
let parens = 0;
let brackets = 0;
let inString = null;
let inComment = false;

for (let i = 0; i < content.length; i++) {
    const char = content[i];
    const next = content[i+1];
    
    if (inComment) {
        if (inComment === '//' && char === '\n') inComment = false;
        if (inComment === '/*' && char === '*' && next === '/') {
            inComment = false;
            i++;
        }
        continue;
    }
    
    if (inString) {
        if (char === inString && content[i-1] !== '\\') inString = null;
        continue;
    }
    
    if (char === '/' && next === '/') { inComment = '//'; continue; }
    if (char === '/' && next === '*') { inComment = '/*'; i++; continue; }
    if (char === "'" || char === '"' || char === '`') { inString = char; continue; }
    
    if (char === '{') braces++;
    if (char === '}') braces--;
    if (char === '(') parens++;
    if (char === ')') parens--;
    if (char === '[') brackets++;
    if (char === ']') brackets--;
    
    if (braces < 0 || parens < 0 || brackets < 0) {
        console.log(`Mismatch at index ${i} (Line roughly ${content.substring(0, i).split('\n').length}): Char ${char}`);
        console.log(`Braces: ${braces}, Parens: ${parens}, Brackets: ${brackets}`);
        // break;
    }
}

console.log('Final counts:');
console.log(`Braces: ${braces}`);
console.log(`Parens: ${parens}`);
console.log(`Brackets: ${brackets}`);
